### ISSUE TITLE
Windows: getCookies return empty list

### ISSUE THREAD

**Issue by ascanime** [pichillilorenzo/flutter_inappwebview#2314]
- [x] I have read the [Getting Started](https://inappwebview.dev/docs/intro/) section
- [x] I have already searched for the same problem

## Environment

| Technology            | Version       |
|-----------------------| ------------- |
| Flutter version       |     3.24.3          |
| Plugin version        |     6.1.3          |
| Android version       |               |
| Windows version       |    11           |
| iOS version           |               |
| macOS version         |               |
| Xcode version         |               |
| Google Chrome version |               |

Device information: 

## Description
I cannot retrieve the list of cookies for a site on my Windows machine

**Expected behavior:** 
should return list of cookies

**Current behavior:** 
return empty list

## Steps to reproduce

- open Inappwebview with url: https://sushiscan.net/ 
- call getCookies() from cookie manager
- return empty list

```
final cookies = await CookieManager.instance()
          .getCookies(url: WebUri("https://sushiscan.net/"));
      print(cookies); // []
```

**Comment by pichillilorenzo** [comment]
Are you using a custom WebViewEnvironment for your WebView? In that case, you need to pass it when you call CookieManager.instance method

**Comment by ascanime** [comment]
Yeah! I passed it but it always return empty list :
 ```
final cookies = await CookieManager.instance(
                            webViewEnvironment: webViewEnvironment)
                        .getCookies(url: WebUri("https://sushiscan.net"));
```

**Comment by pichillilorenzo** [comment]
it seems the problem is related to cookies with partition key.
The actual CookieManager implementation doesn't use the current WebView to get the cookies, so it could be useful add another parameter to `CookieManager.instance` to pass the webview controller to use to get these other cookies.

**Comment by pichillilorenzo** [comment]
The fix will be available in the next beta version `6.2.0-beta.2`.

Here is an example of usage:
```dart
import 'dart:async';
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter_inappwebview/flutter_inappwebview.dart';

WebViewEnvironment? webViewEnvironment;

Future main() async {
  WidgetsFlutterBinding.ensureInitialized();

  if (!kIsWeb && defaultTargetPlatform == TargetPlatform.windows) {
    final availableVersion = await WebViewEnvironment.getAvailableVersion();
    assert(availableVersion != null,
    'Failed to find an installed WebView2 Runtime or non-stable Microsoft Edge installation.');

    webViewEnvironment = await WebViewEnvironment.create(
        settings: WebViewEnvironmentSettings(userDataFolder: 'YOUR_CUSTOM_PATH'));
  }

  runApp(const MaterialApp(home: MyApp()));
}

class MyApp extends StatefulWidget {
  const MyApp({super.key});

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  final GlobalKey webViewKey = GlobalKey();

  InAppWebViewSettings settings = InAppWebViewSettings(
      isInspectable: kDebugMode,
  );
  
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(title: const Text("sushiscan.net test")),
        body: SafeArea(
            child: Column(children: <Widget>[
              Expanded(
                child: Stack(
                  children: [
                    InAppWebView(
                      key: webViewKey,
                      webViewEnvironment: webViewEnvironment,
                      initialSettings: settings,
                      initialUrlRequest: URLRequest(url: WebUri("https://sushiscan.net/")),
                      onLoadStop: (controller, url) async {
                        // get cookie manager
                        final cookieManager = CookieManager.instance(
                            webViewEnvironment: webViewEnvironment);
                        // set also the webViewController to get all the possible cookies
                        print(await cookieManager.getCookies(url: url!, 
                            webViewController: controller));
                      },
                    ),
                  ],
                ),
              ),
            ])));
  }
}
```

![Screenshot 2024-11-12 alle 02 03 16](https://github.com/user-attachments/assets/424f1274-89f1-4cec-92be-45cd12348262)

**Comment by pichillilorenzo** [comment]
Released new version `6.2.0-beta.2` with the fix.
